Chore/ledger trezor update

ivan-angjelkoski opened the pull request.

ivan-angjelkoski committed:
chore: update ledger versions

ivan-angjelkoski committed:
chore: hw wallet derivation path metadata

ivan-angjelkoski committed:
feat: trezor eip1193 provider

ivan-angjelkoski committed:
chore: minor address info

ThomasRalee commented on packages/wallets/wallet-ledger/src/strategy/Ledger/Base.ts:
```diff
@@ -386,6 +408,17 @@ export default class LedgerBase
   ): Promise<LedgerWalletInfo> {
+    // Check metadata first for derivation path
+    if (this.metadata?.derivationPath) {
+      return {
```
Lets make baseDeivationPath a const here

const baseDeviationPath = this.metadata.baseDerivationPath || this.baseDerivationPath,

ThomasRalee commented on packages/wallets/wallet-ledger/src/strategy/Ledger/Base.ts:
```diff
@@ -424,7 +457,7 @@ export default class LedgerBase
     return new LedgerEip1193Provider(this.ledger, {
       chainId: this.evmOptions.evmChainId.toString(),
-      derivationPath: this.baseDerivationPath,
+      derivationPath: this.metadata?.derivationPath,
```
Should we do the same here?

```
 baseDerivationPath: this.metadata.baseDerivationPath || this.baseDerivationPath,
```

ThomasRalee commented on packages/wallets/wallet-ledger/src/strategy/Ledger/Eip1193Provider.ts:
```diff
@@ -21,7 +20,7 @@ export class LedgerEip1193Provider implements Eip1193Provider {
   ) {
     this.ledger = ledger
-    this.derivationPath = "m/44'/60'/0'/0/0"
+    this.derivationPath = params.derivationPath || "m/44'/60'/0'/0/0"
```
lets make `"m/44'/60'/0'/0/0"` a const

ivan-angjelkoski committed:
chore: minor canary

ivan-angjelkoski committed:
chore: minor canary

ivan-angjelkoski committed:
chore: minor canary

ivan-angjelkoski committed:
chore: release

 - @injectivelabs/exceptions@1.16.38-alpha.3
 - @injectivelabs/networks@1.16.38-alpha.2
 - @injectivelabs/sdk-ts@1.16.38-alpha.4
 - @injectivelabs/ts-types@1.16.38-alpha.2
 - @injectivelabs/utils@1.16.38-alpha.3
 - @injectivelabs/wallet-base@1.16.38-alpha.4
 - @injectivelabs/wallet-core@1.16.38-alpha.4
 - @injectivelabs/wallet-cosmos@1.16.38-alpha.4
 - @injectivelabs/wallet-cosmos-strategy@1.16.38-alpha.4
 - @injectivelabs/wallet-cosmostation@1.16.38-alpha.4
 - @injectivelabs/wallet-evm@1.16.38-alpha.4
 - @injectivelabs/wallet-ledger@1.16.38-alpha.4
 - @injectivelabs/wallet-magic@1.16.38-alpha.4
 - @injectivelabs/wallet-private-key@1.16.38-alpha.4
 - @injectivelabs/wallet-strategy@1.16.38-alpha.4
 - @injectivelabs/wallet-trezor@1.16.38-alpha.4
 - @injectivelabs/wallet-turnkey@1.16.38-alpha.4
 - @injectivelabs/wallet-wallet-connect@1.16.38-alpha.4

ivan-angjelkoski committed:
fix: imports proto v2 canary

ivan-angjelkoski committed:
chore: release

 - @injectivelabs/sdk-ts@1.16.38-alpha.5
 - @injectivelabs/utils@1.16.38-alpha.4
 - @injectivelabs/wallet-base@1.16.38-alpha.5
 - @injectivelabs/wallet-core@1.16.38-alpha.5
 - @injectivelabs/wallet-cosmos@1.16.38-alpha.5
 - @injectivelabs/wallet-cosmos-strategy@1.16.38-alpha.5
 - @injectivelabs/wallet-cosmostation@1.16.38-alpha.5
 - @injectivelabs/wallet-evm@1.16.38-alpha.5
 - @injectivelabs/wallet-ledger@1.16.38-alpha.5
 - @injectivelabs/wallet-magic@1.16.38-alpha.5
 - @injectivelabs/wallet-private-key@1.16.38-alpha.5
 - @injectivelabs/wallet-strategy@1.16.38-alpha.5
 - @injectivelabs/wallet-trezor@1.16.38-alpha.5
 - @injectivelabs/wallet-turnkey@1.16.38-alpha.5
 - @injectivelabs/wallet-wallet-connect@1.16.38-alpha.5
